ios/FluentUI.Demo/FluentUIDemoTests/ListItemTest.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import XCTest
+
+class ListItemTest: BaseTest {
+    override var controlName: String { "ListItem" }
+
+    // launch test that ensures the demo app does not crash and is on the correct control page
+    func testLaunch() throws {
+        XCTAssert(app.navigationBars[controlName].exists)
+    }
+
+    func testTitle() throws {
+        let titleElement = app.staticTexts.matching(identifier: "ListItemTitle").firstMatch
+        let textField: XCUIElement = app.textFields.matching(identifier: "titleTextField").firstMatch
+        let newTitle = "A new title"
+
+        clearText(in: textField)
+        textField.typeText(newTitle)
+        XCTAssert(titleElement.label == newTitle, "Title should update when passed in value changes")
+    }
+
+    func testSubtitle() throws {
+        let subtitleElement: XCUIElement = app.staticTexts.matching(identifier: "ListItemSubtitle").firstMatch
+        let textField: XCUIElement = app.textFields.matching(identifier: "subtitleTextField").firstMatch
+        let newSubtitle = "A new subtitle"
+
+        showSubtitleSwitch.tap()
+        XCTAssert(subtitleElement.exists, "Subtitle should appear if a non empty string is passed in")
+
+        clearText(in: textField)
+        textField.typeText(newSubtitle)
+        XCTAssert(subtitleElement.label == newSubtitle, "Subtitle should update when passed in value changes")
+
+        clearText(in: textField)
+        XCTAssertFalse(subtitleElement.exists, "Subtitle should not appear when string is empty")
+    }
+
+    func testFooter() throws {
+        let footerElement: XCUIElement = app.staticTexts.matching(identifier: "ListItemFooter").firstMatch
+        let textField: XCUIElement = app.textFields.matching(identifier: "footerTextField").firstMatch
+        let newFooter = "A new footer"
+
+        XCTAssertFalse(footerElement.exists, "Footer should not appear unless a subtitle is passed in")
+
+        showSubtitleSwitch.tap()
+        showFooterSwitch.tap()
+        XCTAssert(footerElement.exists, "Footer should appear if a non empty string is passed in")
+
+        clearText(in: textField)
+        textField.typeText(newFooter)
+        XCTAssert(footerElement.label == newFooter, "Footer should update when passed in value changes")
+
+        showFooterSwitch.tap()
+        XCTAssertFalse(footerElement.exists, "Footer should not appear when string is empty")
+    }
+
+    func testLeadingContent() throws {
+        let showLeadingContentSwitch: XCUIElement = app.switches.matching(identifier: "leadingContentSwitch").switches.firstMatch
+        let leadingContentElement: XCUIElement = app.images.matching(identifier: "ListItemLeadingContent").firstMatch
+        let leadingContentSizeButton: XCUIElement = app.buttons.matching(identifier: "leadingContentSizePicker").firstMatch
+        let zeroSizeButton: XCUIElement = app.buttons[".zero"].firstMatch
```

Comment:
Any reason why some are `app.buttons.matching` and others are `app.buttons[]`?

Reply:
no they should all be matching, ill update!